atlasdb-commons/src/test/java/com/palantir/common/concurrent/CoalescingSupplierTest.java

```diff
@@ -62,40 +60,38 @@ public class CoalescingSupplierTest {
     private final FreezableSupplier freezableDelegate = new FreezableSupplier(delegate);
     private final CoalescingSupplier<Integer> coalescing = new CoalescingSupplier<>(freezableDelegate);
 
-    private final Supplier<Integer> supplier;
-
-    public CoalescingSupplierTest(String name, Object parameter) {
-        Function<CoalescingSupplierTest, Integer> factory = (Function<CoalescingSupplierTest, Integer>) parameter;
-        supplier = () -> factory.apply(this);
+    public static List<Arguments> getParameters() {
+        return List.of(
+                Arguments.of("blocking", (Function<CoalescingSupplierTest, Integer>) test -> test.coalescing.get()),
+                Arguments.of("async", (Function<CoalescingSupplierTest, Integer>)
+                        test -> unwrap(test.coalescing.getAsync())));
```

Comment:
I think we can actually get rid of the casting by rewriting it like this, which looks cleaner in my opinion:


```
    public static List<Arguments> getParameters() {
        return List.of(
                createArguments("blocking", test -> test.coalescing.get()),
                createArguments("async", test -> unwrap(test.coalescing.getAsync())));
    }

    private static Arguments createArguments(String description, Function<CoalescingSupplierTest, Integer> function) {
        return Arguments.of(description, function);
    }
```